sklearn/metrics/pairwise.py

```diff
@@ -1982,6 +1966,17 @@ def pairwise_distances(
             % (metric, _VALID_METRICS)
         )
 
+    if PairwiseDistances.is_usable_for(X, X if Y is None else Y, metric=metric):
+        # This is an adaptor for one "sqeuclidean" specification.
+        # For this backend, we can directly use "sqeuclidean".
+        if kwds.get("squared", False) and metric == "euclidean":
+            metric = "sqeuclidean"
+            kwds = {}
+
+        return PairwiseDistances.compute(
+            X, X if Y is None else Y, metric=metric, metric_kwargs=kwds
+        )
```

Comment:
And I wonder if we can now remove some parts below that are now unreachable ?

Reply:
code coverage should tell us if any Python part have become unreachable, right? It's weird that is does not show up in the CI status report (maybe because of failing tests?) but here is the report and it seems that we have a bit of dead code:

https://app.codecov.io/gh/scikit-learn/scikit-learn/pull/23958

Reply:
I think parts of the previous implementations will still be used in some context and still will be reachable/covered. For instance boolean metrics aren't yet supported.

Reply:
Moreover, if either `X` or `Y` is F-contiguous, then those implementations aren't usable. 

I think we still need to maintain the old implementations because forcing F-contiguous arrays to be C-contiguous might be costly. What do you think?

Reply:
Further more when inputs are lists of lists of number, we currently need to use the parts bellow because do not get converted to numpy arrays.

See this RFC: https://github.com/scikit-learn/scikit-learn/issues/24745